api/src/main/java/org/openmrs/util/Format.java

```diff
@@ -62,8 +62,25 @@ public static String format(Date date) {
 	public static String format(Date date, FORMAT_TYPE type) {
 		return format(date, Context.getLocale(), type);
 	}
-	
+
+	/**
+	 * This method formats a date object according to a particular locale and returns the date as a string.
+	 * The string can contain only the date (month, day and year), only the time (hours, minutes, seconds) or as a
+	 * timestamp (both date and time).
+	 *
+	 * @param date input date to format as a string
+	 * @param locale input locale to determine how to format the date
+	 * @param type input type to determine how much information from the date is returned
+	 * @return empty string if one of the parameters is null. Otherwise a string object for the date such that it is
+	 * formatted according to locale and the amount of information it contains is determined by type.
+	 * @should not fail with null date
+	 * @should not fail with null locale
```

Comment:
No I did not. However I believe these three annotations are sufficient for the 8 tests that I have added. 3 of the 8 tests test the case when one of these parameters are null. Another 3 of the 8 tests test the case when two of the parameters are null. The remaining two tests test the case when all of them are null or none of them are null. The should annotations I have added implies that regardless of whether any of the parameters are null or not, the method should not fail.